packages/app-builder-lib/src/asar/asarUtil.ts

```diff
@@ -45,237 +43,115 @@ export class AsarPackager {
       fileSets[0],
     ].map(orderFileSet)
 
-    for (const fileSet of orderedFileSets) {
-      unpackedFileIndexMap.set(fileSet, await this.createPackageFromFiles(fileSet, packager.info))
-    }
-    await this.writeAsarFile(orderedFileSets, unpackedFileIndexMap)
-  }
-
-  private async createPackageFromFiles(fileSet: ResolvedFileSet, packager: Packager) {
-    const metadata = fileSet.metadata
-    // search auto unpacked dir
-    const unpackedDirs = new Set<string>()
-    const rootForAppFilesWithoutAsar = path.join(this.destination, "app")
+    const { unpackedPaths, copiedFiles } = await this.detectAndCopy(orderedFileSets, cancellationToken)
+    const unpackGlob = unpackedPaths.length > 1 ? `{${unpackedPaths.join(",")}}` : unpackedPaths.pop()
 
-    if (this.options.smartUnpack !== false) {
-      await detectUnpackedDirs(fileSet, unpackedDirs, this.unpackedDest, rootForAppFilesWithoutAsar)
+    let ordering = this.config.options.ordering || undefined
+    if (!ordering) {
+      // `copiedFiles` are already ordered due to `orderedFileSets` input, so we just map to their relative paths (via substring) within the asar.
+      const filesSorted = copiedFiles.map(file => file.substring(this.rootForAppFilesWithoutAsar.length))
+      ordering = await this.tmpDir.getTempFile({ prefix: "asar-ordering", suffix: ".txt" })
+      await fs.writeFile(ordering, filesSorted.join("\n"))
```

Comment:
Do we need to consider different platforms? On Windows, it should be `\r\n`.

Reply:
Most Windows programs these days can handle LF (`\n`) line endings (even notepad.exe!). It's not a general rule that you should always use CRLF (`\r\n`) on Windows. Honestly, I think the fact that Git for Windows defaults to `autoCRLF=true` is a terrible decision. IMO it's _way_ better to just use LF everywhere, except for like, `.bat` files.